src/main/java/com/iexec/sms/tee/EnableIfTeeProvider.java

```diff
@@ -0,0 +1,70 @@
+package com.iexec.sms.tee;
+
+import com.iexec.common.tee.TeeEnclaveProvider;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+import org.springframework.core.type.AnnotationMetadata;
+
+/**
+ * Define a way to include beans only if a profile is enabled
+ * <p/>
+ * Usage:
+ * Annotate any bean with the following:
+ * {@code
+ * @Conditional(EnableIfTeeProvider.class)
```

Comment:
Can't we use that annotation once (well twice since Gramine and Scone) in a root bean `GramineEnabled`, then only definein others
`@ConditionalOnBean(GramineEnabled.class)`?

Reply:
That's one of the possibilities. However, as you may have read in my last comment, I strive for a more generic architecture, so that we could later add a new provider without having to our base code.

Reply:
Not sure to understand :)
```
@EnableIfTeeProviderDefinition(providers = TeeEnclaveProvider.Gramine)
class GramineEnabled { }

@ConditionalOnBean(GramineEnabled.class)
class  GramineFooService { }
```

```
@EnableIfTeeProviderDefinition(providers = TeeEnclaveProvider.Scone)
class SconeEnabled{ }

@ConditionalOnBean(SconeEnabled.class)
class  SconeFooService { }
```

```
@EnableIfTeeProviderDefinition(providers = TeeEnclaveProvider.Baz)
class BazEnabled{ }

@ConditionalOnBean(BazEnabled.class)
class  BazFooService { }
```
Adding a new provider would look fine don't you think?

Reply:
Sure, but you have to declare a new bean for each provider. That's not a big deal, but that's something I'd like to avoid.